core/story-api/src/main/java/org/lgna/story/implementation/alice/AliceResourceUtilities.java

```diff
@@ -174,6 +177,16 @@ public static TexturedAppearance[] decodeTexture(URL url) {
     }
   }
 
+  // Stretch images to power of two to fix rendering on certain Mac graphics configurations.
+  // The problem was observed specifically with the Baby Penguin model.
+  private static void correctDimensions(TexturedAppearance ta) {
```

Comment:
is it worth checking if the image is already a power of 2?

Reply:
This uses stretchToPowersOfTwo, which does that check first thing, so I did not repeat the test.